### PR TITLE
Bump rails to 7.0.2.3

### PR DESCRIPTION
**Pull request by tvararu**
Fixes https://github.com/DFE-Digital/find-a-lost-trn/security/dependabot/5.

Had to delete `Gemfile.lock` and redo it, which bumped a few other deps, like the linter. Better now than later.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -161,13 +160,12 @@ GEM
       net-protocol
       timeout
     nio4r (2.5.8)
-    nokogiri (1.13.3)
-      mini_portile2 (~> 2.8.0)
+    nokogiri (1.13.3-x86_64-linux)
```
Comment (tvararu) [inline Gemfile.lock:163]: @felixclack this is peculiar to me, the `-linux` suffix

File: Gemfile.lock [hunk 2]
```diff
@@ -284,7 +282,7 @@ GEM
     zeitwerk (2.5.4)
 
 PLATFORMS
-  ruby
+  x86_64-linux
```
Comment (tvararu) [inline Gemfile.lock:285]: This is also peculiar.

File: Gemfile.lock [hunk 3]
```diff
@@ -322,4 +320,4 @@ RUBY VERSION
    ruby 3.0.3p157
 
 BUNDLED WITH
-   2.2.33
+   2.3.8
```
Comment (tvararu) [inline Gemfile.lock:323]: Hopefully not an issue in CI?

**Review by tvararu — COMMENTED** [review]
@felixclack can you run a `bin/bundle` on this locally before we merge to double check that the `-linux` Gemfile.lock updates don't interfere with your setup?

File: Gemfile.lock (hunk 1 above)
Comment (felixclack) [reply Gemfile.lock:163]: I'll check it locally for me

Comment (felixclack) [reply Gemfile.lock:165]: It generated a diff when I ran this locally. I've added a commit to also add the Mac version here.

File: Gemfile.lock (hunk 3 above)
Comment (felixclack) [reply Gemfile.lock:326]: 🤞

**Review by felixclack — APPROVED** [review]
LGTM

File: Gemfile.lock (hunk 1 above)
Comment (tvararu) [reply Gemfile.lock:165]: Nice!

**Pull request merged**